lib/install/management/store_files.go

```diff
@@ -201,20 +214,24 @@ func (d *Dispatcher) getChildren(ds *object.Datastore, dsPath string) ([]string,
 }
 
 // getSortedChildren returns all children under datastore path in reversed order.
-func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string) ([]string, error) {
-	result, err := d.getChildren(ds, dsPath)
+func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string, matchPattern ...string) ([]string, error) {
+	result, err := d.getChildren(ds, dsPath, matchPattern...)
 	if err != nil {
 		return nil, err
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(result)))
 	return result, nil
 }
 
-func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
+func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string, matchPattern ...string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
 	defer trace.End(trace.Begin(dsPath))
 
+	// match every subfolers by default
+	if len(matchPattern) == 0 {
+		matchPattern = append(matchPattern, "*")
+	}
 	spec := types.HostDatastoreBrowserSearchSpec{
-		MatchPattern: []string{"*"},
+		MatchPattern: matchPattern,
```

Comment:
though according to the API, I feel like you're right, dev/abc would be returned and removed. I need to figure out why .iso and tether.debug are not removed. Is this the default behavior when --force is not specified?